test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,633 @@
+
+        logger.info("=== Test Case 3 Completed ===\n")
+
+    def test_retrieval_with_valid_inputs(self):
+        """Test case 4: Valid customer_guid and chat_id"""
+        logger.info("=== Starting Test Case 4: Valid customer_guid and chat_id ===")
+
+        # Add multiple chat messages to the chat
+        chat_url = f"{self.BASE_URL}/chat"
+        messages_to_add = [
+            "Message 1: You will get the correct answer once AI is integrated.",
+            "Message 2: You will get the correct answer once AI is integrated.",
+            "Message 3: You will get the correct answer once AI is integrated.",
+
+        ]
+
+        for message in messages_to_add:
+            message_data = {
+                "customer_guid": self.valid_customer_guid,
+                "question": message  # Using 'question' to represent the message
+            }
+            logger.info(f"INPUT: Creating message with data: {str(message_data)}")
+            message_response = requests.post(chat_url, json=message_data)
+            logger.info(f"OUTPUT: Message creation response status: {message_response.status_code}")
+            self.assertEqual(message_response.status_code, 200, "Failed to create test message.")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,
+            "page": 1,
+            "page_size": 10
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Verify that the response status code is 200
+        self.assertEqual(response.status_code, 200, "Expected status code 200")
+
+        messages = response.json().get('messages')
+        self.assertIsInstance(messages, list)  # Check that messages is a list
+
+        # Verify that the chat_id and customer_guid in the response match the input
+        if messages:  # Check if messages list is not empty
+            # Assuming the first message corresponds to the chat_id and customer_guid
+            first_message = messages[0]
+            self.assertEqual(first_message['chat_id'], self.valid_chat_id,
+                             "Chat ID in response does not match the input chat ID")
+            self.assertEqual(first_message['customer_guid'], self.valid_customer_guid,
+                             "Customer GUID in response does not match the input customer GUID")
+
+        # Check for reverse chronological order
+        for i in range(len(messages) - 1):
+            current_timestamp = messages[i]['timestamp']
+            next_timestamp = messages[i + 1]['timestamp']
+            self.assertGreaterEqual(current_timestamp, next_timestamp,
+                                    "Messages are not in reverse chronological order")
+
```

Comment:
We should also verify whether the input questions you send are arriving as output and you should also verify whether user or system has sent the message

Reply:
Good

Reply:
Reopen